### PR TITLE
update plugin registration

### PR DESCRIPTION
**Pull request by prolic**
see: https://github.com/prooph/event-store/issues/232

### CODE REVIEW

**Comment by prolic** [comment]
ping @codeliner

**Comment by prolic** [comment]
@oqq review?

File: src/CausationMetadataEnricher.php [hunk 1]
```diff
@@ -14,30 +14,35 @@
...
+     * @var array
+     */
+    private $messageBusListeners = [];
+
+    public function attachToEventStore(ActionEventEmitterEventStore $eventStore): void
+    {
+        $this->eventStoreListeners[] = $eventStore->attach(
             ActionEventEmitterEventStore::EVENT_APPEND_TO,
             function (ActionEvent $event): void {
                 if (null === $this->currentCommand || ! $this->currentCommand instanceof Message) {
```
Comment (oqq) [inline src/CausationMetadataEnricher.php:48]: `null` is also NOT an instance of `Message`, so this is a doubled expression I guess. Not sure about performance here.

File: src/CausationMetadataEnricher.php [hunk 2]
```diff
@@ -57,7 +62,7 @@ function (ActionEvent $event): void {
             1000
         );
 
-        $eventEmitter->attachListener(
+        $this->eventStoreListeners[] = $eventStore->attach(
             ActionEventEmitterEventStore::EVENT_CREATE,
             function (ActionEvent $event): void {
                 if (null === $this->currentCommand || ! $this->currentCommand instanceof Message) {
```
Comment (oqq) [inline src/CausationMetadataEnricher.php:68]: `null` is also NOT an instance of `Message`, so this is a doubled expression I guess. Not sure about performance here.

Comment (prolic) [reply src/CausationMetadataEnricher.php:68]: done, see https://github.com/prooph/event-store-bus-bridge/pull/25

File: src/CausationMetadataEnricher.php (hunk 1 above)
Comment (prolic) [reply src/CausationMetadataEnricher.php:48]: done, see https://github.com/prooph/event-store-bus-bridge/pull/25

**Pull request merged**